libraries/bezier-rs/src/bezier/transform.rs

```diff
@@ -201,29 +201,37 @@ impl Bezier {
 			// Perform no processing on the subcurve if it's already scalable.
 			if subcurve.is_scalable() {
 				result_beziers.push(subcurve);
-				result_t_values.push(t_subcurve_end);
+				result_t_values.push([t_subcurve_start, t_subcurve_end]);
 				return;
 			}
 
 			// Greedily iterate across the subcurve at intervals of size `step_size` to break up the curve into maximally large segments
 			let mut segment: Bezier;
 			let mut t1 = 0.;
 			let mut t2 = step_size;
+			let mut is_prev_valid = false;
 			while t2 <= 1. + step_size {
 				segment = subcurve.trim(TValue::Parametric(t1), TValue::Parametric(f64::min(t2, 1.)));
 				if !segment.is_scalable() {
 					t2 -= step_size;
 
 					// If the previous step does not exist, the start of the subcurve is irreducible.
 					// Otherwise, add the valid segment from the previous step to the result.
-					if f64::abs(t1 - t2) >= step_size {
+					if is_prev_valid {
 						segment = subcurve.trim(TValue::Parametric(t1), TValue::Parametric(t2));
-						result_beziers.push(segment);
-						result_t_values.push(t_subcurve_start + t2 * (t_subcurve_end - t_subcurve_start));
+						if segment.is_scalable() {
+							result_beziers.push(segment);
+							result_t_values.push([t_subcurve_start + t1 * (t_subcurve_end - t_subcurve_start), t_subcurve_start + t2 * (t_subcurve_end - t_subcurve_start)]);
+						} else {
+							t2 = t1 + step_size;
+						}
 					} else {
-						return;
+						t2 = t1 + step_size;
 					}
```

Comment:
You should probably be able to combine these else cases.

Reply:
To combine them, I'd want to refactor as:
```
if is_prev_valid && segment.is_scalable() { .. }
```
However, that would involve computing `subcurve.trim(...)` every time despite the fact that it isn't always necessary. Is this alright to do?

Reply:
I noticed it might require rearrangements that would hurt performance, but I thought I'd ask in case you could find another less obvious way that didn't occur to me. If you think this is the only way to arrange that logic without causing extra computational cost, it's fine to leave it as is.